### PR TITLE
Added function to remove a form group as suggested in Issue #2294

### PR DESCRIPTION
**Pull request by Brainshack**
As suggested in Issue #2294, I created a function to remove a form group

### CODE REVIEW

**Comment by core23** [comment]
:+1:

**Comment by Brainshack** [comment]
Just to clearifiy: The failed CI builds, seem to fail because of Unit tests, that arent even a part of this PR. My tests worked.

**Comment by Soullivaneuh** [comment]
Thanks for this PR.

Two things:

* Could you please squash your commit on rebase it from master in order to re-run test with fresh code?
* Could you please add some notes about that on documentation?

**Comment by Brainshack** [comment]
So I did as you requests. Wasn't sure where to put the docs, so I just made a new Cookbook Recipe. Hope that's okay.

**Comment by Brainshack** [comment]
Is there a way to trigger the one failing CI build? It seems like to have had connection issues with github. This should (finally) run through, and seeing this one build beeing red, gives me the chills. :D

File: Form/FormMapper.php [hunk 1]
```diff
@@ -173,6 +173,43 @@ public function remove($key)
...
+
+        if (false !== $key) {
+            unset($tabs[$tab]['groups'][$key]);
+        }
+        if ($deleteTabWhenEmpty && count($tabs[$tab]['groups']) == 0) {
+            unset($tabs[$tab]);
+        }
+
+        $this->setTabs($tabs);
+        $this->setGroups($groups);
```
Comment (Soullivaneuh) [inline Form/FormMapper.php]: Should returns itself IMO.

Comment (Brainshack) [inline Form/FormMapper.php]: Done

**Comment by Soullivaneuh** [comment]
Note: You should not merge upstream/master but rebase it instead. :+1: 

Can you please squash your commits? Will test it later.

**Comment by Brainshack** [comment]
Well dang, sorry about that. Not really sure how to do that though, now that the merge has beend pushed? :(

**Comment by Soullivaneuh** [comment]
Follow [this sample](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) but use `fixup` keywork instead of `squash`.

**Comment by Brainshack** [comment]
Well, the problem is, not that I merged the upstream, there is a whole lot of commits between my latests, and and the last one I did about a month ago. If I understand correctly, squash and fixup will combine the commit with the previous one, right?

**Comment by Brainshack** [comment]
Okay, I think I got it this time. Git can get tricky from time to time.

**Comment by borodiliz** [comment]
@Brainshack @Soullivaneuh Any update on this? Very interesting IMO. Thanks!

**Comment by OskarStark** [comment]
This branch has to be rebased

**Comment by core23** [comment]
@Brainshack Please rebase. I'm waiting for this feature!

**Comment by OskarStark** [comment]
@core otherwise just for this and keep the commits of @Brainshack and provide a new PR

**Comment by OskarStark** [comment]
Applied in c969be0fcd91b6ee7bb82540d84f696ca28fa783

Thank you @core23 for rebasing an creating a new PR :+1:

**Pull request closed without merging**